lightning/src/offers/invoice_request.rs

```diff
@@ -322,12 +322,30 @@ impl InvoiceRequest {
 		self.signature
 	}
 
+	/// Creates an [`Invoice`] for the request with the given required fields and using the
+	/// [`Duration`] since [`std::time::SystemTime::UNIX_EPOCH`] as the creation time.
+	///
+	/// See [`InvoiceRequest::respond_with_no_std`] for further details where the aforementioned
```

Comment:
It's been a while since I last checked this PR, so I hope I'm not relitigating something that's already been discussed, but I am wondering:

Do we wanna hide respond_with behind std, or do we wanna modify respond_with_no_std to make its timestamp argument an Option, so respond_with would always be available, but depending on the std setting, would or would not check for expiry?

Reply:
This isn't checking for expiry; it's constructing the required invoice creation time. We conditionally check that the underlying offer has not expired when calling `InvoiceBuilder::build`, though.